### ISSUE TITLE
binary-compat: rust: signal() error

### ISSUE THREAD

**Issue by razvand** [unikraft/unikraft#490]
### Describe the bug

When running [rust static PIE binaries](https://github.com/unikraft/static-pie-apps/tree/master/lang/rust) under binary compatibility mode, using [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader), it errs when using `signal`:

```
fatal runtime error: assertion failed: signal(libc::SIGPIPE, libc::SIG_IGN) != libc::SIG_ERR
```

### Steps to reproduce

Use [`run-app-elfloader`](https://github.com/unikraft/run-app-elfloader) to run the [`rust` static PIE binary](https://github.com/unikraft/static-pie-apps/tree/master/lang/rust). Use the command below, assuming the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader) and the [`static-pie-apps` repository](https://github.com/unikraft/static-pie-apps/) are cloned in the same directory:

```
$ ./run_elfloader ../static-pie-apps/lang/rust/helloworld-gnu
```

The output shows the Unikraft boot message together with the error message on `signal`:

```
Powered by
o.   .o       _ _               __ _
Oo   Oo  ___ (_) | __ __  __ _ ' _) :_
oO   oO ' _ `| | |/ /  _)' _` | |_|  _)
oOo oOO| | | | |   (| | | (_) |  _) :_
 OoOoO ._, ._:_:_,\_._,  .__,_:_, \___)
            Mimas 0.7.0~1e17a9c8-custom
fatal runtime error: assertion failed: signal(libc::SIGPIPE, libc::SIG_IGN) != libc::SIG_ERR
```

To build your own setup, including Unikraft and aim to fix this bug, see [these instructions](https://unikraft.org/community/hackathons/2022-05-lyon/bincompat/#00-setup).

### Expected behavior

The `helloworld-gnu` and `helloworld-musl` applications should start under binary compatibility mode and print `Hello, world!`.

### Which architectures were you using or does this bug affect?

x86_64

### Which platforms were you using or does this bug affect?

kvm

**Comment by razvand** [comment]
@StefanJum , should this be closed since #494 was merged?

**Comment by StefanJum** [comment]
@razvand yes, this was closed by #314
